core/src/main/java/com/linecorp/armeria/common/logging/LoggingDecoratorBuilder.java

```diff
@@ -36,6 +36,15 @@
     private LogLevel requestLogLevel = LogLevel.TRACE;
     private LogLevel successfulResponseLogLevel = LogLevel.TRACE;
     private LogLevel failedResponseLogLevel = LogLevel.WARN;
+    private Function<? super RequestLog, LogLevel> requestLogLevelMapper =
+            log -> requestLogLevel();
+    private Function<? super RequestLog, LogLevel> responseLogLevelMapper =
+            log -> log.responseCause() == null ? successfulResponseLogLevel() : failedResponseLogLevel();
+    private boolean isSetRequestLogLevel;
+    private boolean isSetSuccessfulResponseLogLevel;
+    private boolean isSetFailedResponseLogLevel;
```

Comment:
I think we can integrate these two flags into one.
`isSetSuccessfulResponseLogLevel`, `isSetFailedResponseLogLevel` ->
`isResponseLogLevelSet`

Reply:
Looks good. But how do I raise a error message in `responseLogLevelMapper()`. 
In the method, it check 2 conditions about `{Successful|Failed} ResponseLogLevel`.
If the user set the response level only one (e.g success or failed), I don't know what the property is set.  If I'll change this, the error message is not helpful to user I think. Is this good?

Reply:
If we set `isResponseLogLevelSet` to `true` when either `(Successful|Failed)ResponseLogLevel` is called, we can just check `isResponseLogLevelSet` in `responseLogLevelMapper` and raise an exeption like `throw new IllegalStateException("successfulResponseLogLevel or failedResponseLogLevel has been set already.");`.

Reply:
or just `responseLogLevel has been set already.` would be enough. 😄

Reply:
I'm fine with either options. If we merge the flags, we could update the exception message, like `successfulResponseLogLevel or failedResponseLogLevel have been set already.`

Reply:
I think "successfulResponseLogLevel or failedResponseLogLevel has been set already." is more clear to me if I change it.

Reply:
OK. let me change it.

Reply:
Fixed